### PR TITLE
Frontend: Server-side filtering of executions by status

### LINKED ISSUE #396: Support the filtration by PASS and FAILED from ExecutionView and HistoryView
1) When you click on the number of FAILED tests on the History view, you should get only results for failed tests.
2) You should be able to sort tests

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Change endpoints to retrieve test executions filtered by status
* Change links from `HistoryView` to lead to filtered test executions
* Fix `TopBar`, which displayed query string before
* Added filtering in the table
* Update `build_and_test.yml` for correct codecov uploading

image

Closes #396

### CODE REVIEW

**Review by kgevorkyan — APPROVED** [review]

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt [hunk 1]
```diff
@@ -70,10 +76,12 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
      * Returns number of TestExecutions with this [executionId]
      *
      * @param executionId an ID of Execution to group TestExecutions
+     * @param status
      */
-    @GetMapping("/testExecutionsCount")
-    fun getTestExecutionsCount(@RequestParam executionId: Long) =
-            testExecutionService.getTestExecutionsCount(executionId)
+    @Suppress("KDOC_WITHOUT_RETURN_TAG")  // https://github.com/cqfn/diKTat/issues/965
```
[the remark this answers is not in the document]
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt]: Must've been by accident. Removed suppress.

**Comment by akuleshov7** [comment]
I love this!

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/TopBar.kt [hunk 2]
```diff
@@ -76,6 +76,7 @@ fun topBar() = fc<TopBarProps> { props ->
                     }
                 }
                 location.pathname
+                    .substringBeforeLast("?")
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/TopBar.kt:79]: what's for?

Comment (petertrr) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/TopBar.kt:79]: Previously query string would be displayed in the `TopBar` along with the rest of the URL. Now it'll be trimmed.

**Pull request merged**